projects/driver_controls/src/direction_fsm.c

```diff
@@ -0,0 +1,89 @@
+#include "direction_fsm.h"
+#include "input_event.h"
+#include "event_arbiter.h"
+#include "log.h"
+
+// Direction selector FSM state definitions
+
+FSM_DECLARE_STATE(state_neutral);
+FSM_DECLARE_STATE(state_forward);
+FSM_DECLARE_STATE(state_reverse);
+
+// Direction selector FSM transition table definitions
+
+FSM_STATE_TRANSITION(state_neutral) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_DIRECTION_SELECTOR_DRIVE, state_forward);
+  FSM_ADD_TRANSITION(INPUT_EVENT_DIRECTION_SELECTOR_REVERSE, state_reverse);
+}
+
+FSM_STATE_TRANSITION(state_forward) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_DIRECTION_SELECTOR_NEUTRAL, state_neutral);
+  FSM_ADD_TRANSITION(INPUT_EVENT_DIRECTION_SELECTOR_REVERSE, state_reverse);
+}
+
+FSM_STATE_TRANSITION(state_reverse) {
+  FSM_ADD_TRANSITION(INPUT_EVENT_DIRECTION_SELECTOR_DRIVE, state_forward);
+  FSM_ADD_TRANSITION(INPUT_EVENT_DIRECTION_SELECTOR_NEUTRAL, state_neutral);
+}
+
+// Direction selector FSM arbiter functions
+
+static bool prv_check_neutral(const Event *e) {
+  return !(e->id == INPUT_EVENT_PEDAL_COAST || e->id == INPUT_EVENT_PEDAL_PRESSED);
+}
+
+static bool prv_check_forward(const Event *e) {
+  return (e->id != INPUT_EVENT_POWER);
+}
+
+static bool prv_check_reverse(const Event *e) {
+  return !(e->id == INPUT_EVENT_POWER || e->id == INPUT_EVENT_CRUISE_CONTROL);
+}
+
+// Direction selector FSM output functions
+
+static void prv_state_neutral(FSM *fsm, const Event *e, void *context) {
+  EventArbiterCheck *event_check = fsm->context;
+  *event_check = prv_check_neutral;
+
+  InputEventData *data = &e->data;
+  data->components.state = 0;
```

Comment:
It would be nice if these state values were enums or at least defined somewhere. I'd suggest defining them in the header and just including the header within your CAN FSM.